radio/apps/schedules/models.py

```diff
@@ -227,45 +291,58 @@ def between(cls, after, before, exclude=None, live=False, schedule_board=None):
+        list_schedules = (
+            cls.objects.filter(
+                programme__start_date__lte=dt,
+                programme__end_date__isnull=True) |
+            cls.objects.filter(
+                programme__start_date__lte=dt,
+                programme__end_date__gte=dt))
         if exclude:
             list_schedules = list_schedules.exclude(id=exclude.id)
-        list_schedules = list_schedules.select_related('programme', 'schedule_board')
+        list_schedules = list_schedules.select_related(
+            'programme', 'schedule_board')
         earlier_date = None
         earlier_schedule = None
         for schedule in list_schedules:
-            # if schedule != exclude:
             date = schedule.date_before(dt)
             if date and (earlier_date is None or date > earlier_date):
                 earlier_date = date
                 earlier_schedule = schedule
-        if earlier_schedule is None or dt > earlier_date + earlier_schedule.runtime():  # Todo: check
+        if (earlier_schedule is None or
+                dt > earlier_date + earlier_schedule.runtime()):
+            # XXX Todo: check
             return None, None
         return earlier_schedule, earlier_date
 
     @classmethod
     def get_next_date(cls, programme, after):
         list_schedules = cls.objects.filter(programme=programme, type='L')
-        list_schedules = list_schedules.filter(programme__end_date__isnull=True) | list_schedules.filter(programme__end_date__gte=after)
-        list_schedules = list_schedules.filter(schedule_board__start_date__isnull=False, schedule_board__end_date__isnull=True) | list_schedules.filter(schedule_board__end_date__gte=after)
-        # list_schedules = list_schedules.select_related('programme', 'schedule_board', 'source__programme', 'source__schedule_board')
+        list_schedules = (
+            list_schedules.filter(programme__end_date__isnull=True) |
+            list_schedules.filter(programme__end_date__gte=after))
+        list_schedules = (
+            list_schedules.filter(
+                schedule_board__start_date__isnull=False,
+                schedule_board__end_date__isnull=True) |
+            list_schedules.filter(schedule_board__end_date__gte=after))
+
         closer_date = None
         closer_schedule = None
         for schedule in list_schedules:
-            # if schedule != exclude:
             date = schedule.date_after(after)
             if date and (closer_date is None or date < closer_date):
                 closer_date = date
                 closer_schedule = schedule
-        if closer_schedule is None:  # Todo: check
+        if closer_schedule is None:  # XXX Todo: check
             return None, None
         return closer_schedule, closer_date
 
     def __unicode__(self):
-        return self.get_day_display() + ' - ' + self.start_hour.strftime('%H:%M')
+        return ' - '.join(
```

Comment:
Whoops, while writing some unit tests it emerged that this is not working. We need a sequence. I'll provide a fix.